### ISSUE TITLE
Manually Gettting Region Data?

### ISSUE THREAD

**Issue by freeradicalx** [mathuin/TopoMC#26]
Hey there, first of all amazing tool! I'll probably just be using this to make a geographically accurate map of my old hometown for friends and I to play Minecraft in, but I'm floored by it's potential for non-gameplay uses like visualization, analysis, etc. So thank you!

Like MinorMajor in the previous thread I spent several hours last night tracking down and installing pyrex, nbt, cython, suds, PyYAML, GDAL, scipy, numpy and gFortran in order to get TopoMC running on OS X (10.7.5) only to be presented with the "No products are available for this location" error, which lead me to your issue ticket regarding the USGS's unfortunate service shutdown.

But I still really want to see this work, as it seems that any data pulled down from USGS by GetRegion.py is also available for email delivery via the map viewer at http://viewer.nationalmap.gov/viewer/. Is this the data that GetRegion.py needs? And if so, is there a way to manually format and present this data to GetRegion.py for processing? Are there any other data sources that are accessed, as well?

Basically, is there a way to manually bridge the gap created by USGS's service shutdown?

**Comment by mathuin** [comment]
The short answer is: yes, if you are really, really good with GDAL command lines and reading the TopoMC source code, you could pull it off.  Request elevation and landcover for the desired area area via the map viewer, trim both regions to be precisely the correct dimensions, put them in the right directories, and modify TopoMC to skip all the download stuff.

That being said, if you could read the TopoMC source code that well, you could probably write Python to download and trim, and then shoehorn it into TopoMC so everything works for everyone. ;-)  That's what I plan to do the next time I get eight straight hours to work on the problem.

Jack.

**Comment by freeradicalx** [comment]
Yikes, I don't even know Python, let alone GDAL. The task at hand is certainly out of my league. May your upcoming weeks be graced with plentiful free time, sir! =P

I'm curious as to how a new python downloader would grab USGS data, now that they've apparently disabled dynamic downloads and the only way to grab elevation/landcover I can see is to check out a cart of products that get delivered as links via email. Would a new downloader need to work interactively with their http map viewer as well as receive / open their email? Sounds fantastically complicated.

**Comment by mathuin** [comment]
Last I checked, the web services still offered downloads just not of arbitrary regions.  So you ask for a small area and get the entire tile of that area via download instead of only what you requested. The new way saves the USGS disk space and compute time but wastes bandwidth in my opinion. I will update the other ticket with my progress.

**Comment by mathuin** [comment]
The new hotfix works with the USGS servers to access tiled files.  You can download these files automatically with ./GetRegion.py or you can download them by hand and place them in the Downloads directory, in subdirectories named after their product keys (N3F or N1F for elevation and L6N or L1L for landcover at the moment), where ./GetRegion.py can find them.  So you got what you wanted after all!

**Comment by freeradicalx** [comment]
Amazing! Can't wait to get home and give it a try! I'll let you know how it goes.
